### PR TITLE
General DOC updates, cleanup and refactor.

### PR DESCRIPTION
**Pull request by lgirdwood**
Mostly moving documents and topics into the correct category, also filling in missing section and creating place holders for new docs.

Signed-off-by: Liam Girdwood <liam.r.girdwood@linux.intel.com>

### CODE REVIEW

File: introduction/index.rst [hunk 1]
```diff
@@ -4,192 +4,155 @@ Introduction to the SOF Project
...
-  the src/arch directory to your new architecture. The code in the
-  architecture directory mainly deals with architecture abstraction and
-  initialization of any architecture IP like MMU, IRQs and caches
-  alongside providing optimized version of common C functions (memcpy,
-  memset, etc) for that architecture. Adding a new architecture also
-  usually means adding a new host platform too.
-
-How do I add support for host platform X?
+Who is working on |SOF| ?
+  Proffesional developers from a number of companies (please check the git
```
Comment (intelkevinputnam) [inline introduction/index.rst]: Line 142: "Professional ..."

File: introduction/index.rst [hunk 2]
```diff
@@ -4,192 +4,155 @@ Introduction to the SOF Project
...
-#. **Crosstool-NG toolchain.** Crosstool-NG is used to build a GNU cross
-   toolchain (gcc, gdb, binutils, etc.) that is used to build the firmware
-   binaries. Other compilers and toolchains can also be used to build the
-   firmware.
+SDK Introduction
+================
 
-#. **Qemu DSP and host emulator.** Qemu is used to provide a functional
-   emulator to simultaneously trace and debug driver and DSP firmware code.
+The |SOF| SDK comprises of the following ingredients:
```
Comment (intelkevinputnam) [inline introduction/index.rst]: Line 36: "... SDK is comprised of ..."

File: introduction/index.rst [hunk 3]
```diff
@@ -4,192 +4,155 @@ Introduction to the SOF Project
...
+infrastructure and SDK. SOF provides infrastructure, real-time control pieces,
+and audio drivers as a community project. The project is governed by the |SOF|
+|TSC| (TSC) who are prominent and active developers from the community.
+SOF is developed in public and hosted on the github platform.
 
 The firmware and SDK are intended for developers who are interested in
 audio or signal processing on modern DSPs or who are interested in
-microkernels that run on small but powerful processors.
+microkernels that run on small but powerful processors. SOF provides a
+framework where audio developers can create, test and tune :-
```
Comment (intelkevinputnam) [inline introduction/index.rst]: Line 15: Do no follow colon ":" with hyphen "-"

File: introduction/index.rst [hunk 4]
```diff
@@ -4,192 +4,155 @@ Introduction to the SOF Project
...
+as a stand alone firmware. SOF host drivers currently supports Linux OS today. 
 
-#. **DSP Platform Driver.** The platform driver is a platform specific
-   driver that abstracts the low level platform DSP hardware into a common
-   generic API that is used by the upper layers. This includes code that
-   will initialize the DSP and boot the firmware.
+The SOF driver has a modular stack based architecture that is dual licensed
+BSD & GPL code allowing it to be ported to other OSes and RTOSes. 
 
+The host driver is responsible for :-
```
Comment (intelkevinputnam) [inline introduction/index.rst]: Line 70: remove hyphen "-"

File: introduction/index.rst [hunk 5]
```diff
@@ -4,192 +4,155 @@ Introduction to the SOF Project
...
+
+Do I need to open source my driver code changes ?
+  No for the bottom two layers of the driver stack. i.e. if you are porting the 
+  driver to another OS, these changes can be kept private. Please note that the
+  driver GPL source files are all Linux specific and should not be ported to
+  another OS.
+
+How can I get involved ?
+  There are several ways to get involved, only the github method is mandatory :-
+#. Join github and fork the project.
```
Comment (intelkevinputnam) [inline introduction/index.rst]: Line 130-132: The numbered list needs to follow a carriage return and be indented to match the rest of the definition list content.

File: architectures/host/index.rst [hunk 6]
```diff
@@ -3,4 +3,47 @@
...
+
+	`Sound Open Firmware Driver Architecture`
+
+The right-hand side of the diagram shows the mailbox/doorbell mechanism and the DSP.
+
+The PCM and IPC drivers can be reused without modification on every
+platform. The platform differentiation will occur via the topology data
+and firmware. There is also scope for differentiation via the machine
+driver and platform driver. The ACPI or Device Tree could be used to
+specify the HW configuration.
```
Comment (plbossart) [inline architectures/host/index.rst]: we should probably call out that the ACPI/DT stuff is more static/hardware-specific by nature whereas a topology can be regenerated with the firmware on the same hardware?

File: architectures/index.rst [hunk 7]
```diff
@@ -3,13 +3,15 @@
 Supported Architectures
 #######################
 
-SOF project supports multiple different architectures.
+SOF is intended to run on many different hardware architectures and therefore
+it's not tightly or even loosely coupled to any particular DSP or host hardware
+architecture. The TSC ensure that any DSP or host architecture specific code
```
Comment (plbossart) [inline architectures/index.rst]: define TSC.

File: introduction/index.rst (hunk 3 above)
Comment (plbossart) [reply introduction/index.rst]: he's been doing this for as long as I can remember, must be some secret encoding

File: architectures/index.rst [hunk 8]
```diff
@@ -3,13 +3,15 @@
...
+is partitioned to reside in architecture specific directories with generic
```
Comment (plbossart) [inline architectures/index.rst]: architecture-specific?

File: introduction/index.rst [hunk 9]
```diff
@@ -4,192 +4,155 @@ Introduction to the SOF Project
...
-  work.
+The Linux SOF ALSA/ASoC driver is upstream in Linux v5.2 onwards.
+
+
+Firmware Toolchain
+------------------
+
+GNU GCC can be used as a free SOF compiler alongside proprietary DSP vendor
+compilers. The choice of compiler is up to the user depending on features
+and budget. GCC complier is free.
```
Comment (plbossart) [inline introduction/index.rst]: free is repeated here, not necessary. Also it's open-source, not free we should be using.

File: introduction/index.rst [hunk 10]
```diff
@@ -4,192 +4,155 @@ Introduction to the SOF Project
...
-  architectures like ARM, MIPS etc.
+General FAQ
+===========
 
-How can I get involved?
-  Please join the developer mailing where new development features and
-  patches are discussed:
-  http://alsa-project.org/mailman/listinfo/sound-open-firmware
+What license does the firmware use ?
+  The firmware is released using a standard BSD 3 clause license with some
```
Comment (plbossart) [inline introduction/index.rst]: 3-clause

File: introduction/index.rst [hunk 11]
```diff
@@ -4,192 +4,155 @@ Introduction to the SOF Project
...
-  |SOF| has a similar development model to the Linux kernel. Patches are
-  discussed and posted on the mailing list before being merged. The
+Do I need to open source my firmware code changes ?
+  No. The firmware BSD and MIT licensed code means you can keep code
+  changes private. Patches are always welcomed if do decide to open source
+  work.
+
+What license does the host driver use ?  
+  Most of the host driver code is dual licensed BSD or GLPLv2 only 
+  (users choice). The part of the driver that is GPLv2 only is the Linux
```
Comment (plbossart) [inline introduction/index.rst]: user's?

File: architectures/index.rst (hunk 7 above)
Comment (plbossart) [reply architectures/index.rst]: or use |TSC|

File: introduction/index.rst (hunk 3 above)
Comment (lgirdwood) [reply introduction/index.rst]: long time guilty of this :)

**Comment by lgirdwood** [comment]
@intelkevinputnam @deb-intel ping. Updated and now ready for merging. I have further updates to go on top of this.

**Comment by lgirdwood** [comment]
@intelkevinputnam @deb-intel I've merged, but please do feel free to incrementally correct my grammar.

**Comment by deb-intel** [comment]
Thanks, @lgirdwood. I hope to look at this today or tomorrow. :)

**Comment by lgirdwood** [comment]
@deb-intel thanks and appreciated. Please can you publish after making any corrections.

**Comment by deb-intel** [comment]
Ok.

**Pull request merged**